### ISSUE TITLE
DOC003 (style mismatch) not reported when it should be

### ISSUE THREAD

**Issue by mzebrak** [jsh9/pydoclint#244]
Hi, I noticed that pydoclint doesn't detect style mismatches when it should.

Version:
```
pydoclint, version 0.6.6
```

Consider the following settings:
```toml
 [tool.pydoclint]
 style = "google"
 arg-type-hints-in-docstring = false
 check-return-types = false
 check-yield-types = false
 check-style-mismatch = true
```

And this is considered an invalid style:

```python
def add(a: float, b: float) -> float:
    """
    Add two numbers.

    Args:
    -----
    a: First number.
    b: Second number.

    Returns:
    --------
    The sum of the two numbers.

    Examples:
    ---------
    >>> add(1, 2)
    3
    """
    return a + b
```

because it raises a bunch of errors (including `DOC003`):

```
    1: DOC001: Function/method `add`: Potential formatting errors in docstring. Error message: No specification for "Args": "" (Note: DOC001 could trigger other unrelated violations under this function/method too. Please fix the docstring formatting first.)
    1: DOC003: Function/method `add`: Docstring style mismatch. (Please read more at https://jsh9.github.io/pydoclint/style_mismatch.html ). You specified "google" style, but the docstring is likely not written in this style.
    1: DOC101: Function `add`: Docstring contains fewer arguments than in function signature.
    1: DOC103: Function `add`: Docstring arguments are different from function arguments. (Or could be other formatting issues: https://jsh9.github.io/pydoclint/violation_codes.html#notes-on-doc103 ). Arguments in the function signature but not in the docstring: [a: float, b: float].
    1: DOC201: Function `add` does not have a return section in docstring
```

while this is OK for pydoclint:
```python
def add(a: float, b: float) -> float:
    """
    Add two numbers.

    Args:
        a: First number.
        b: Second number.

    Returns:
    --------
    The sum of the two numbers.

    Examples:
    ---------
    >>> add(1, 2)
    3
    """
    return a + b
```

and gives no errors.

Expected behaviour:
pydoclint would force me to use such a formatting

```python
def add(a: float, b: float) -> float:
    """
    Add two numbers.

    Args:
        a: First number.
        b: Second number.

    Returns:
        The sum of the two numbers.

    Examples:
        >>> add(1, 2)
        3
    """
    return a + b
```

**Comment by jsh9** [comment]
Hi! This is an edge case, and I've fixed it in the most recent PR.
